### PR TITLE
Add Input Validation for Token Address

### LINKED ISSUE #4: Add Input Validation for Token Address
**Description**: Enhance token address validation in distribute functions.

**Tasks**:
- [ ] Add ERC20 interface check for token address
- [ ] Validate token decimals
- [ ] Add revert messages for invalid tokens

Here's where the token input validation should take place:

```rust 
     fn distribute(
            ref self: ContractState,
            amount: u256,
            recipients: Array<ContractAddress>,
            token: ContractAddress,
        ) {}
```

and this function:

```rust
        fn distribute_weighted(
            ref self: ContractState,
            amounts: Array<u256>,
            recipients: Array<ContractAddress>,
            token: ContractAddress,
        ) {}
```

Make sure you test that the implementation works for edge-cases by writing test for the it.

## Application before the commencement of the ODweek will not be considered.

### PR DESCRIPTION
**Pull request by Amarjeet325**
close #4

### CODE REVIEW

File: src/distribute.cairo [hunk 1]
```diff
@@ -118,6 +118,13 @@ mod Distributor {
 
     #[abi(embed_v0)]
     impl DistributorImpl of IDistributor<ContractState> {
+        fn validate_token(self: @ContractState, token: ContractAddress) {
```
Comment (Utilitycoder) [inline src/distribute.cairo:121]: I think this should be an internal function. I do not see any need for it to be external. Please fix that!

**Review by Utilitycoder — CHANGES_REQUESTED** [review]
